pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -811,26 +806,22 @@ func (p *PriorityQueue) done(pod types.UID) {
 	// remove events which is only referred from this Pod
 	// so that the receivedEvents map doesn't grow infinitely.
 	event := p.receivedEvents.Front()
-	if inFlightPod.previousEvent != nil {
+	if inFlightPod.previousEvent != nil && inFlightPod.previousEvent.Next() != nil {
```

Comment:
I think just this was right: `if inFlightPod.previousEvent != nil`

if `inFlightPod.previousEvent.Next() == nil` it means that there were no events after the pod was poped.

Please add a unit test for this case.

Reply:
Ah... So there are cases we need to care about:

- The previousEvent is already removed from the list.
  - In this case, `inFlightPod.previousEvent.Next()` would be nil. I have `inFlightPod.previousEvent.Next() != nil` for this case.
- previousEvent is actually the latest event in receivedEvents.
  - In this case, the current logic is wrong because we don't need to handle any events in this case.

Let me have a fix.